includes/class-coinsnap-bitcoin-donation-public-donors.php

```diff
@@ -0,0 +1,284 @@
+            'hierarchical'       => false,
+            'supports'           => ['title'],
+            'show_in_rest'       => true
+        ]);
+    }
+
+    public function register_custom_meta_fields()
+    {
+        register_meta('post', '_coinsnap_bitcoin_donation_donor_name', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_amount', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_message', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_form_type', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_email', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_address', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_payment_id', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+
+        register_meta('post', '_coinsnap_bitcoin_donation_custom_field', [
+            'object_subtype' => 'bitcoin-pds',
+            'type' => 'string',
+            'single' => true,
+            'show_in_rest' => true,
+        ]);
+    }
+
+    public function add_public_donors_metaboxes()
+    {
+        add_meta_box(
+            'coinsnap_bitcoin_donation_public_donors_details',
+            'Donor Details',
+            [$this, 'render_public_donors_metabox'],
+            'bitcoin-pds',
+            'normal',
+            'high'
+        );
+    }
+
+    public function render_public_donors_metabox($post)
+    {
+        wp_nonce_field('coinsnap_bitcoin_donation_public_donors_nonce', 'coinsnap_bitcoin_donation_public_donors_nonce');
+
+        $name = get_post_meta($post->ID, '_coinsnap_bitcoin_donation_donor_name', true);
+        $amount = get_post_meta($post->ID, '_coinsnap_bitcoin_donation_amount', true);
+        $message = get_post_meta($post->ID, '_coinsnap_bitcoin_donation_message', true);
+        $form_type = get_post_meta($post->ID, '_coinsnap_bitcoin_donation_form_type', true);
```

Comment:
The idea for this field is to use one shortcode, and not to show the different shortcode based on type.

So we will have one shortcode and render template based on the selection of this field.

Reply:
Keeping the current functionality.